scripts/exregional_run_prdgen.sh

```diff
@@ -0,0 +1,364 @@
+export OMP_STACKSIZE=${OMP_STACKSIZE_RUN_PRDGEN}
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+eval ${PRE_TASK_CMDS}
+
+nprocs=$(( NNODES_RUN_PRDGEN*PPN_RUN_PRDGEN ))
+if [ -z "${RUN_CMD_PRDGEN:-}" ] ; then
+  print_err_msg_exit "\
+  Run command was not set in machine file. \
+  Please set RUN_CMD_PRDGEN for your platform"
+else
+  print_info_msg "$VERBOSE" "
+  All executables will be submitted with command \'${RUN_CMD_PRDGEN}\'."
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Get the cycle date and hour (in formats of yyyymmdd and hh, respectively)
+# from CDATE.
+#
+#-----------------------------------------------------------------------
+#
+yyyymmdd=${PDY}
+hh=${cyc}
+#-----------------------------------------------------------------------
+#
+# A separate ${post_fhr} forecast hour variable is required for the post
+# files, since they may or may not be three digits long, depending on the
+# length of the forecast.
+#
+#-----------------------------------------------------------------------
+#
+
+len_fhr=${#fhr}
+if [ ${len_fhr} -eq 9 ]; then
+  post_min=${fhr:4:2}
+  if [ ${post_min} -lt 15 ]; then
+    post_min=00
+  fi
+else
+  post_min=00
+fi
+
+subh_fhr=${fhr}
+if [ ${len_fhr} -eq 2 ]; then
+  post_fhr=${fhr}00
+elif [ ${len_fhr} -eq 3 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    post_fhr="${fhr:1}00"
+  else
+    post_fhr=${fhr}00
+  fi
+elif [ ${len_fhr} -eq 9 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:1:2}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:1:2}${fhr:4:2}"
+    fi
+  else
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:0:3}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:0:3}${fhr:4:2}"
+    fi
+  fi
+else
+  print_err_msg_exit "\
+The \${fhr} variable contains too few or too many characters:
+  fhr = \"$fhr\""
+fi
+
+# replace fhr with subh_fhr
+echo "fhr=${fhr} and subh_fhr=${subh_fhr}"
+fhr=${subh_fhr}
+
+#
+# background files
+#
+prslev=${COMOUT}/${NET}.${cycle}${dot_ensmem}.prslev.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+natlev=${COMOUT}/${NET}.${cycle}${dot_ensmem}.natlev.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+testbed=${COMOUT}/${NET}.${cycle}${dot_ensmem}.testbed.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+ififip=${COMOUT}/${NET}.${cycle}${dot_ensmem}.ififip.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
+
+# extract the output fields for the testbed files
+touch ${testbed}
+if [[ ! -z ${TESTBED_FIELDS_FN} ]]; then
+  if [[ -f "${PARMdir}/upp/${TESTBED_FIELDS_FN}" ]]; then
+    wgrib2 ${prslev} | grep -F -f "${PARMdir}/upp/${TESTBED_FIELDS_FN}" | wgrib2 -i -grib ${testbed} ${prslev}
+  else
+    echo "${PARMdir}/upp/${TESTBED_FIELDS_FN} not found"
+  fi
+fi
+if [[ ! -z ${TESTBED_FIELDS_FN2} ]]; then
```

Comment:
This is a bug if `TESTBED_FIELDS_FN2` is not defined? I think @gspetro-NOAA also poointed this out.

Reply:
Good catch, I missed that somehow.

Reply:
Also both of these files are not available in current SRW app. So we would need to have those two fix files incorporated.
```
  CUSTOM_POST_CONFIG_FP: '{{ [user.PARMdir, "upp/postxconfig-NT-fv3lam_rrfs.txt"]|path_join }}'
  CUSTOM_POST_PARAMS_FP: '{{ [user.PARMdir, "upp/params_grib2_tbl_new"]|path_join }}'
  TESTBED_FIELDS_FN: "testbed_fields_bgdawp.txt"
  TESTBED_FIELDS_FN2: "testbed_fields_bgrd3d.txt"
```
Moreover, these variables are first used by the `run_post` job so they should be defined there.
Another reason why order of tasks is important.

Reply:
I was thinking about the best way to handle this too.  I tested the prdgen task on WCOSS2 without these files and it completed successfully for me.  If the files aren't defined in the config.yaml file, then the testbed file generation part of the job should be skipped.  But once the run_post job is updated and the necessary files are added, this logic will already be in place for the testbed file generation and it can be turned on then.